examples.md

```diff
@@ -170,6 +180,33 @@ The yarn cache directory will depend on your operating system and version of `ya
       ${{ runner.os }}-yarn-
 ```
 
+## OCaml/Reason - esy
+Esy allows you to export built dependencies and import pre-built dependencies.
+```yaml
+    - name: Restore Cache
+      id: restore-cache
+      uses: actions/cache@v1
+      with:
+        path: _export
+        key:  ${{ runner.os }}-esy-${{ hashFiles('esy.lock/index.json') }}
+        restore-keys: |
+          ${{ runner.os }}-esy-
+    - name: Esy install
```

Comment:
Not familiar with this ecosystem, why do you import the cache *after* installing?

Reply:
`esy install` will mostly resolve the dependency graph and prepare the terrain, this usually takes about ~20s. It's the `esy build` phase that is expensive and that will take between 30seconds and up to an hour or even more on some projects. (For example, building Z3 takes ~20mn, building Coq takes ~30mn). From a clean start, it will also build OCaml from scratch (~10mn).

The `import-dependencies` command looks into the `_export` folders and gets the built packages that *it needs*. In order to know what it needs, it needs to resolve the dependencies first and that is done by `esy install`, so it has to be done before importing the pre-built dependencies.